### ISSUE TITLE
Weird exception message when table don't exists

### ISSUE THREAD

**Issue by rufanov** [nette/database#79]
Hello! 

A friend of mine created simple application that use nette/database. In controller action he has this code:

    $clients = $db->table('clients');
    $client = $db->get($id);

At first time it worked fine. But later he started to want to rename 'clients' to 'users' . So he changed table name in database, and changed code:

    $users = $db->table('users');
    $user = $db->get($id);

Seems fine, but.. He got exception:

> LogicException
> Table 'users' does not have a primary key.

What?!? He looked at database structure, and said "but i **have** primary key in this table!". 

Then he looked at *Nette.Database.Structure* cache and found real problem - there is no any data about table named 'users'. So he removed this cache to rebuild it, and after this application started to work fine again.

I tried to use table that really don't exists in database:
 
    $db->table('foobar')->get(1);

and got same exception:

> Table 'foobar' does not have a primary key.

This is very strange behavior to show this exception message, isn't? I think if table actually exists in database but not present in cache, then *nette/database* should rebuild cache. Or, if not, then it should throw more meaningful exception like "Table don't exists".

**Comment by semerak-michal** [comment]
Zdravím,

myslím, že úpravy pro toto issue omylem způsobily, že se vyhazuje výjimka tam kde dřív ne.

Metoda `resolveFQTableName` v `Nette\Database\Strucure` se volá z metody `getPrimaryKey` téže třídy. Problém nastane ve chvíli, když je argumentem pohled.

Pohledy se v metodě loadStructure přeskakují:

```php
foreach ($structure['tables'] as $tablePair) {
    if ($tablePair['view']) {
        continue;
    }
    ....
```

a díky tomu v `$this->structure['columns']` ani `$this->structure['primary']` nikdy nebude a vyskočí ona vyjímka.

**Comment by janfiedler** [comment]
Potvrzuji, zrovna jsem prováděl update z 2.3.4 na 2.3.5 a začalo mi to vyhazovat ty výjimky u dat, které jsou vyčítány z pohledu.

**Comment by dg** [comment]
English, please.

**Comment by f3l1x** [comment]
@semerak-michal @janfiedler @dg 

In case of view, there's always exception (Table '*' does not exist).

**Comment by Unlink** [comment]
It should be fixed in 87a60dcb7857012cd6c297be544d76d28f2da003

**Comment by f3l1x** [comment]
@Unlink Thanks. It's fixed in 2.3.6, right?
